storage/client_status_test.go

```diff
@@ -0,0 +1,221 @@
+// and Val are at least the expected value.  The latest actual stats are
+// returned.
+func compareStoreStatus(t *testing.T, store *storage.Store, expectedStoreStatus *proto.StoreStatus) *proto.StoreStatus {
+	storeStatusKey := engine.StoreStatusKey(int32(store.Ident.StoreID))
+	gArgs, gReply := getArgs(storeStatusKey, 1, store.Ident.StoreID)
+	if err := store.ExecuteCmd(gArgs, gReply); err != nil {
+		t.Fatalf("failure getting store status: %s", err)
+	}
+	if gReply.Value == nil {
+		t.Errorf("could not find store status at: %s", storeStatusKey)
+	}
+	storeStatus := &proto.StoreStatus{}
+	if err := gogoproto.Unmarshal(gReply.Value.GetBytes(), storeStatus); err != nil {
+		t.Errorf("could not unmarshal store status: %+v", gReply)
+	}
+	if expectedStoreStatus.StoreID != storeStatus.StoreID {
+		t.Errorf("actual store ID does not match expected\nexpected: %+v\nactual: %v\n", expectedStoreStatus, storeStatus)
+	}
+	if expectedStoreStatus.NodeID != storeStatus.NodeID {
+		t.Errorf("actual node ID does not match expected\nexpected: %+v\nactual: %v\n", expectedStoreStatus, storeStatus)
+	}
+	if expectedStoreStatus.RangeCount != storeStatus.RangeCount {
+		t.Errorf("actual RangeCount does not match expected\nexpected: %+v\nactual: %v\n", expectedStoreStatus, storeStatus)
+	}
+	if storeStatus.Stats.LiveBytes < expectedStoreStatus.Stats.LiveBytes {
+		t.Errorf("actual Live Bytes is not greater or equal to expected\nexpected: %+v\nactual: %v\n", expectedStoreStatus, storeStatus)
+	}
+	if storeStatus.Stats.KeyBytes < expectedStoreStatus.Stats.KeyBytes {
+		t.Errorf("actual Key Bytes is not greater or equal to expected\nexpected: %+v\nactual: %v\n", expectedStoreStatus, storeStatus)
+	}
+	if storeStatus.Stats.ValBytes < expectedStoreStatus.Stats.ValBytes {
+		t.Errorf("actual Val Bytes is not greater or equal to expected\nexpected: %+v\nactual: %v\n", expectedStoreStatus, storeStatus)
+	}
+	if storeStatus.Stats.LiveCount < expectedStoreStatus.Stats.LiveCount {
+		t.Errorf("actual Live Count is not greater or equal to expected\nexpected: %+v\nactual: %v\n", expectedStoreStatus, storeStatus)
+	}
+	if storeStatus.Stats.KeyCount < expectedStoreStatus.Stats.KeyCount {
+		t.Errorf("actual Key Count is not greater or equal to expected\nexpected: %+v\nactual: %v\n", expectedStoreStatus, storeStatus)
+	}
+	if storeStatus.Stats.ValCount < expectedStoreStatus.Stats.ValCount {
+		t.Errorf("actual Val Count is not greater or equal to expected\nexpected: %+v\nactual: %v\n", expectedStoreStatus, storeStatus)
+	}
+	return storeStatus
+}
+
+// TestStoreStatus checks the store status after each range scan to ensure that
+// it is being updated correctly.
+func TestStoreStatus(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	store, stopper := createTestStoreWithScanner(t)
+	defer stopper.Stop()
+	splitKey := proto.Key("b")
+	content := proto.Key("test content")
+
+	// This is set to 2 times that scan interval.
+	waitDuration := 20 * time.Millisecond
```

Comment:
For replication tests I added a ForceReplicationScan method. Maybe we should do the same here instead of running the scanner in the background and sleeping (this would also make the test constructor question moot).